src/goto-symex/symex_complexity.cpp

```diff
@@ -0,0 +1,195 @@
+/*******************************************************************\
+
+Module: Symbolic Execution
+
+Author: John Dumbell
+
+\*******************************************************************/
+
+#include "symex_complexity.h"
+#include "goto_symex_state.h"
+#include <cmath>
+
+symex_complexityt::symex_complexityt(
+  message_handlert &logger,
+  const optionst &options)
+  : log(logger)
+{
+  std::size_t limit = options.get_signed_int_option("symex-complexity-limit");
+  if((complexity_active = limit > 0))
+  {
+    // This gives a curve that allows low limits to be rightly restrictive,
+    // while larger numbers are very large.
+    max_complexity = static_cast<std::size_t>((limit * limit) * 25);
```

Comment:
Can this overflow?

Reply:
Hard to do on mine, but technically it can. I just assumed people wouldn't give it a number  _that_ high.